.github/workflows/knit.yml

```diff
@@ -0,0 +1,37 @@
+name: Verify Docs Code Snippets
+
+on:
+  push:
+    branches:
+      - main
+  workflow_dispatch:  # Manual trigger
```

Comment:
since this check is fast, I'd suggest moving the job to check.yml, see https://github.com/JetBrains/koog/blob/73975a5c040f89b7f131c7de28df98d44142aa4f/.github/workflows/checks.yml#L44-L43

alternatively, you can create a new workflow with docs and knit, and run them only if /docs is changed

Reply:
Not applicable as decided to make it as a pr check